ownCloud/Client/ClientQueryViewController.swift

```diff
@@ -389,30 +404,36 @@ extension ClientQueryViewController : OCQueryDelegate {
 		query.requestChangeSet(withFlags: OCQueryChangeSetRequestFlag(rawValue: 0)) { (_, changeSet) in
 			DispatchQueue.main.async {
 				self.items = changeSet?.queryResult
-				self.tableView.reloadData()
 
 				switch query.state {
-					case .contentsFromCache, .idle:
-						if self.items?.count == 0 {
-							self.message(show: true, imageName: "folder", title: "Empty folder".localized, message: "This folder contains no files or folders.".localized)
+				case .contentsFromCache, .idle:
+					if self.items?.count == 0 {
+						if self.searchController.searchBar.text != "" {
+							self.message(show: true, imageName: "folder", title: "No matches".localized, message: "There is no results for this search".localized)
 						} else {
-							self.message(show: false)
+							self.message(show: true, imageName: "folder", title: "Empty folder".localized, message: "This folder contains no files or folders.".localized)
 						}
+					} else {
+						self.message(show: false)
+						self.tableView.reloadData()
+					}
 
-					case .targetRemoved:
-						self.message(show: true, imageName: "folder", title: "Folder removed".localized, message: "This folder no longer exists on the server.".localized)
+				case .targetRemoved:
+					self.message(show: true, imageName: "folder", title: "Folder removed".localized, message: "This folder no longer exists on the server.".localized)
+					self.tableView.reloadData()
 
-					default:
-						self.message(show: false)
+				default:
+					self.message(show: false)
+					self.tableView.reloadData()
```

Comment:
I'm not sure if calling `reloadData` is needed here.

Reply:
It was called always for all the cases in line 392 before. What I did is to take in mind the search and not update the `tableView` for this specific use-case and maintain the call in the rest of the cases in the switch statement.